scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -259,24 +259,35 @@ class FormatWriter(formatOps: FormatOps) {
     }
     initStyle.newlines.alwaysBeforeTopLevelStatements && {
       val formatToken = toks(i).formatToken
+
       def checkPackage: Option[Boolean] =
         if (!initStyle.activeForEdition_2019_11) None
         else
           Some(owners(formatToken.left))
             .collect { case term: Term.Name => term.parent }
             .flatten
-            .collect { case pkg: Pkg => pkg.stats.headOption }
+            .collect {
+              // package a
+              case pkg: Pkg =>
+                pkg.stats.headOption
+
+              // package a.b.c
+              case select: Term.Select =>
```

Comment:
would this pattern match `package a.b.c {` as well? there might an `Apply` there somewhere.